hearth/lib/hearth/telemetry/base.rb

```diff
@@ -36,7 +43,7 @@ class TracerBase
       # lifecycle of the Span and its parent manually.
       #
       # @param [String] name Span name
-      # @param [optional Context] with_parent Parent Context
+      # @param [optional object] with_parent Parent Context
```

Comment:
You should remove "optional" from these - yardoc doesn't care about this, only about the types. Why is Object correct? Is it because it can be arbitrary? This should also be Object not object if so.